src/main/scala/cognite/spark/v1/SdkV1Rdd.scala

```diff
@@ -53,18 +54,21 @@ final case class SdkV1Rdd[A, I](
     val groupedStreams = streams.grouped(config.parallelismPerPartition).toSeq
     val currentStreamsAsSingleStream = groupedStreams(split.index).reduce(_.merge(_))
 
-    val processedIds = new ConcurrentHashMap[I, Unit]
-    val processChunk = (chunk: Chunk[A]) => {
-      chunk.filter { i =>
-        // putIfAbsent returns null if the key did not exist, in which ase we
-        // should keep (and process) the item.
-        Option(processedIds.putIfAbsent(uniqueId(i), ())).isEmpty
-      }
+    val processChunk = if (deduplicateRows) {
```

Comment:
I think it would be good if we could try only this change first, instead of changing several things at once. Since ids for RAW are larger than for other resource types (being strings and all), it seems reasonable that this might be the issue.

Reply:
Ok, I'll separate it. It just took me few attempt to find it and all included changes seem to help